src/main/java/com/wb/service/WeatherService.java

```diff
@@ -0,0 +1,49 @@
+package com.wb.service;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.wb.payload.WeatherMapper;
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@Slf4j
+public class WeatherService {
```

Comment:
Brakuje testów jednostkowych do tej klasy a jest ona ważna. Musisz zamockować `RestTemplate` tak by zwracał różne scenariusze i czy Twój serwis sobie z nimi poradzi. Tu jest przyzwoity przykład https://www.baeldung.com/restclienttest-in-spring-boot oczywiście to jeden ze sposobów. Nie potrzebujesz nawet Springa do tego, Samo mockito by wystarczyło